### PR TITLE
Modify color utility classes in night mode dashboards

### LINKED ISSUE #14326: Pivot tables don’t look right in dark mode (embedded preview)
**Describe the bug**
Pivot table visualization doesn't look right in dark mode in embedded previews. It seems that there is no CSS dealing with this context.

**Logs**
N/A

**To Reproduce**
1. Create a question with pivot table visualization and save it (adding it to the dashboard or not doesn't make any difference)
2. Click on sharing icon, then choose "Sharing and embedding"
3. Embed this question/dashboard in an application
4. In the right sidebar choose Dark mode
5. Note that cells' background color is hardcoded and doesn't adjust to the dark mode

**Expected behavior**
There should be a style specific to the dark mode.

**Screenshots**
image

**Severity**
Functionality doesn't seem to be affected, but it looks weird and it's really hard to read some fields.

**Additional Context**
`frontend/src/metabase/css/core/colors.css`

```css
.bg-light, .bg-light-hover:hover {
    background-color: #f9fbfc;
}

.bg-medium, .bg-medium-hover:hover {
    background-color: #edf2f5;
}

.text-dark {
    color: #4c5773;
}

.text-medium {
    color: #949aab;
}
```

It seems that (at least when it comes to the text color) there was some CSS dealing with the night mode but it's being overwritten here:
image

Also, note that CSS is included twice - reported in #13219

### PR DESCRIPTION
**Pull request by kdoh**
Closes #14326 

## What this does
- Modifies some of our `bg-*` and `border-*` util classes to use dark colors when in night mode. I went with the "modify the css classes themselves" approach instead of adding different classes in the viz type itself based on an `isNightmode` to make sure this would apply to other visualizations in the future as well.

I don't love this as a long term solution, since it doesn't cover every possible case or class, but it helps clean up the big legibility issue with our new PivotTables when in night mode.

Showing the effect on pivot tables

Before:
image

After:
image

### CODE REVIEW

**Review by paulrosenzweig — APPROVED** [review]
Looks good!

File: frontend/src/metabase/css/dashboard.css [hunk 1]
```diff
@@ -83,6 +83,25 @@
   fill: currentColor;
 }
 
+.Dashboard.Dashboard--night .bg-light {
+  background-color: var(--color-bg-black);
+}
+
+.Dashboard.Dashboard--night .bg-medium {
+  background-color: #596269;
```
Comment (nemanjaglumac) [inline frontend/src/metabase/css/dashboard.css:91]: Do we have any similar value stored in css vars?

Comment (kdoh) [reply frontend/src/metabase/css/dashboard.css:91]: I don't believe so. I had initially tried one of our vars but nothing was suitable. I don't know that this needs to truly be a var if it's only used here.

Comment (nemanjaglumac) [reply frontend/src/metabase/css/dashboard.css:91]: True. If we end up needing it elsewhere in the future, we can make a new var then.

**Comment by nemanjaglumac** [comment]
I couldn't test this before embed endpoints PR is merged, so I added `.Dashboard--night` to the "regular" dashboard. Not sure if that gives me the correct picture, but just in case I want to share something:

Note how card title is still really dark.
image

image

**Comment by kdoh** [comment]
@nemanjaglumac I'm not able to reproduce that. It's possible once the embed stuff lands we might need to make more adjustments, but I'd rather do that once we know for sure than do anything preemptively.

**Comment by nemanjaglumac** [comment]
Absolutely agreed.
Great work fixing this so quickly! 💪

**Pull request merged**